internal/pkg/cli/deploy/env.go

```diff
@@ -100,11 +141,115 @@ func NewEnvDeployer(in *NewEnvDeployerInput) (*envDeployer, error) {
 			Env:             in.Env,
 		},
 		newStackSerializer: func(in *deploy.CreateEnvironmentInput, lastForceUpdateID string, oldParams []*awscfn.Parameter) stackSerializer {
-			return stack.NewEnvConfigFromExistingStack(in, lastForceUpdateID, oldParams)
+			return cfnstack.NewEnvConfigFromExistingStack(in, lastForceUpdateID, oldParams)
+		},
+		envDescriber:      envDescriber,
+		envManagerSession: envManagerSession,
+		newLBDescriber: func(sess *session.Session) lbDescriber {
+			return elbv2.New(sess)
+		},
+		newServiceStackDescriber: func(svc string, sess *session.Session) stackDescriber {
+			return stack.NewStackDescriber(cfnstack.NameForService(in.App.Name, in.Env.Name, svc), sess)
 		},
 	}, nil
 }
 
+// Verify checks that the manifest configuration is valid to deploy alongside
+// the currently deployed resources.
+func (d *envDeployer) Verify(ctx context.Context, mft *manifest.Environment) error {
+	if mft.CDNEnabled() && mft.HTTPConfig.Public.Certificates == nil && d.app.Domain != "" {
```

Comment:
nit: I'm a little worried that we're "leaking" the internal structure of our manifest to packages outside of `manifest` (like in `stack` and `deploy`) that would make it difficult to change the internal representation.
 
What do you think of adding a method instead?
```suggestion
	if mft.CDNEnabled() && !mft.HasImportedPublicALBCerts() && d.app.Domain != "" {
```

Reply:
just a random thought -  I think I did like how `isManagedCDNEnabled` characterized this conditional and how it helped with understanding the flow. What about ```if managedCDNEnabled :=  mft.CDNEnabled() && !mft.HasImportedPublicALBCerts() && d.app.Domain != ""; managedCDNEnabled {```